mlflow/store/cli.py

```diff
@@ -0,0 +1,101 @@
+                       "artifact in a directory this way.")
+@click.option("--local-dir", "-l", required=True,
+              help="Directory of local artifacts to log")
+@click.option("--run-id", "-r", required=True,
+              help="Run ID into which we should log the artifact.")
+@click.option("--artifact-path", "-a",
+              help="If specified, we will log the artifact into this subdirectory of the " +
+                   "run's artifact directory.")
+def log_artifacts(local_dir, run_id, artifact_path):
+    """
+    Creates a new experiment in the configured tracking server.
+    """
+    store = _get_store()
+    artifact_uri = store.get_run(run_id).info.artifact_uri
+    artifact_repo = ArtifactRepository.from_artifact_uri(artifact_uri, store)
+    artifact_repo.log_artifacts(local_dir, artifact_path)
+    eprint("Logged artifact from local dir %s to artifact_path=%s" % (local_dir, artifact_path))
+
+
+@commands.command("list",
+                  help="Return all the artifacts directly under run's root artifact " +
+                       "directory, or a sub-directory. The output is a JSON-formatted list.")
+@click.option("--run-id", "-r", required=True,
+              help="Run ID to be listed")
+@click.option("--artifact-path", "-a",
+              help="If specified, a path relative to the run's root directory to list.")
+def list_artifacts(run_id, artifact_path):
+    """
+    Creates a new experiment in the configured tracking server.
+    """
+    artifact_path = artifact_path if artifact_path is not None else ""
+    store = _get_store()
+    artifact_uri = store.get_run(run_id).info.artifact_uri
+    artifact_repo = ArtifactRepository.from_artifact_uri(artifact_uri, store)
+    file_infos = artifact_repo.list_artifacts(artifact_path)
+    print(_file_infos_to_json(file_infos))
+
+
+def _file_infos_to_json(file_infos):
+    json_list = [message_to_json(file_info.to_proto()) for file_info in file_infos]
+    return "[" + ", ".join(json_list) + "]"
+
+
+@commands.command("download-artifacts",
+                  help="Download an artifact file or directory to a local directory. " +
+                       "The output is the name of the file or directory on the local disk.")
+@click.option("--run-id", "-r", required=True,
+              help="Run ID from which to download")
+@click.option("--artifact-path", "-a",
+              help="If specified, a path relative to the run's root directory to download")
+def download_artifacts(run_id, artifact_path):
+    """
+    Creates a new experiment in the configured tracking server.
+    """
+    artifact_path = artifact_path if artifact_path is not None else ""
+    store = _get_store()
+    artifact_uri = store.get_run(run_id).info.artifact_uri
+    artifact_repo = ArtifactRepository.from_artifact_uri(artifact_uri, store)
+    artifact_location = artifact_repo.download_artifacts(artifact_path)
```

Comment:
We should make sure that this doesn't delete the temp dir it downloads to on exit, if some of the stores ever do that (I doubt it but maybe add a comment in the doc of artifact_repo).

Reply:
All have remove_on_exit=False, added to the docs.